### PR TITLE
Improvements in CMAKE

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
Use `IS_DIRECTORY` rather than `EXISTS`
Also fixes populate to make it possible to rebuild targets that use local source
AND removes the un-necessary linux specific SVN source for chibios.

## Motivation and Context
Improves check to make it consistant.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by josesimoes** [comment]
Looks good! Any chance you can replace all the similar ones in other platforms  CMakes?

**Comment by josesimoes** [comment]
@networkfusion I was experimenting with this and I think we can improve it even further!
The reason for those checks if the directory existed and it had content date back from the very beginning. Seems that the call to `FetchContent_Declare` performs those checks and outputs a clear enough message.

Here's the output when pointed to an empty directory or when the path doesn't exist:
```
[cmake] CMake Error at C:/Program Files/CMake/share/cmake-3.22/Modules/ExternalProject.cmake:2882 (message):
[cmake]   No download info given for 'ti_xdctools-populate' and its source directory:
[cmake] 
[cmake]    E:/GitHub/TI_XDCTools
[cmake] 
[cmake]   is not an existing non-empty directory.  Please specify one of:
[cmake] 
[cmake]    * SOURCE_DIR with an existing non-empty directory
[cmake]    * DOWNLOAD_COMMAND
[cmake]    * URL
[cmake]    * GIT_REPOSITORY
[cmake]    * SVN_REPOSITORY
[cmake]    * HG_REPOSITORY
[cmake]    * CVS_REPOSITORY and CVS_MODULE
``` 

I guess we can make that simplification and just remove that outer IF and make is even simpler.

**Comment by robin-jones** [comment]
I also tried the `FetchContent_Declare` and ran into issues with CMAKE background building the folders source, which runs into errors in some of them, so might need to be careful there... But if you can make it work, I have nothing against it.

**Comment by networkfusion** [comment]
@josesimoes , if I have done it correctly in the last commit (for TI) I will also go through the others...

File: targets/ChibiOS/CMakeLists.txt [hunk 1]
```diff
@@ -416,24 +394,22 @@ if(NF_FEATURE_USE_SPIFFS)
...
-                GIT_TAG "nf-build"
-            )
+        message(STATUS "SPIFFS source from: ${SPIFFS_SOURCE}")
 
-        else()
-            message(FATAL_ERROR "Couldn't find SPIFFS source at ${SPIFFS_SOURCE}/")
-        endif()
+        FetchContent_Declare(
+            spiffs
+            GIT_REPOSITORY ${SPIFFS_SOURCE}
```
Comment (josesimoes) [inline targets/ChibiOS/CMakeLists.txt:401]: this should be SOURCE_DIR too

Comment (networkfusion) [reply targets/ChibiOS/CMakeLists.txt]: Done.

**Review by josesimoes — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**